### PR TITLE
Show information about a single change request

### PR DESCRIPTION
**Pull request by ddaze**
* Use `repoload show <change-id>` to display information about a CR.
  This will show the git project, branch, change subject, owner,
  commit message, the comments and more.
  (Not implemented are the comments on individual files.)

* Remove the `:` in the repoload change view `<change-id: <Name>`.
  This allows simpler parsing of the <change-id> and also looks nicer.

This new feature allows to show a simple UI to browse the open CRs
with fzf.

Signed-off-by: David Zerulla <ddaze@outlook.de>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +129,16 @@ is to *not* use repoload with non-UTF8 character sets.
 
 It's planned to fix this issue in the next version of repoload.
 
+### Interactive change view with fzf
+
+With fzf (A command-line fuzzy finder) it's easy to build a simple CR viewer.
+To browse all open CR's load them with `repoload changes` and use fzf's preview
+function to call `repoload show` for the selected CR.
+
+Install fzf: https://github.com/junegunn/fzf
```
Comment (lengfeld) [inline README.md:138]: I think it's good style to have a descriptive sentence/paragraph before every code block that acts as a introduction for the following commands. So can you move this `Install fzf ...` belwo the code block and have a small text paragraph here.

File: repoload/repoload.py [hunk 2]
```diff
@@ -74,9 +85,15 @@ def get_gerrit_url():
...
+def query(url: str, query: str, port: int = 29418,
+          options: Union[None, GerritOptions, Sequence[GerritOptions]] = None):
+    opt = ""
+    if isinstance(options, list):
+        opt = " ".join([o.value for o in options])
+    elif isinstance(options, GerritOptions):
+        opt = options.value
+
+    cmd = "ssh -p {port} {url} gerrit query --format=JSON {options} '{query}'".format(port=port, url=url, options=opt, query=query)
     p = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
```
Comment (lengfeld) [inline repoload/repoload.py:97]: It's time to get rid of this white space splitting with `cmd.split()` which starts to break when an option or query contains a whitespace itself. That's an issue in every shell implementation and usage.

Since we have python here and `Popen` accepts a list of strings, the code should also create a list of strings instead of relying on the no whitespace in the options hope. It will break someday.

So `cmds` should be a list of strings and elements should be appended with the list functions `.append()` or `.extend()`.

In short python is programming, not shell scripting :-)

**Review by lengfeld — COMMENTED** [review]
> This allows simpler parsing of the <change-id> and also looks nicer.

This sentence from the commit message is debatable. Now a human reader cannot easily distinguish between the Change id and the subject line. Maybe the subject line starts with a number.

Is there also another option instead of just removing the colon?

File: repoload/repoload.py [hunk 3]
```diff
@@ -151,13 +168,84 @@ def get_changes_for_topic(topic):
...
+    print("Created at\t{}".format(datetime.fromtimestamp(c["createdOn"])))
+    print("Last updated\t{}".format(datetime.fromtimestamp(c["lastUpdated"])))
+    print("Url\t\t{}".format(c["url"]))
+
+    print("Change subject:\t{}".format(c["subject"]))
+    print("Change owner:\t{} ({})".format(c["owner"]["name"], c["owner"]["email"]))
+
+    print('\nCommit Message:\n')
+    for line in str.splitlines(c["commitMessage"]):
+        print(w.fill(line))
```
Comment (lengfeld) [inline repoload/repoload.py:208]: At least for the commit message I don't think it's useful to line wrapping it in the repolint tool. The commit message is normally already correctly line wrapped. Only long lines like URLs, compile output or log message are sometimes longer than 72 characters.

I think it's better to not modify the commit message to have a consistent and easier to understand behavior of repolint.

File: repoload/repoload.py [hunk 4]
```diff
@@ -151,13 +168,84 @@ def get_changes_for_topic(topic):
...
+    print('Comments:')
+    w.initial_indent = "\t> "
+    w.subsequent_indent = "\t> "
+    w.break_long_words = False
+    w.break_on_hyphens = False
+    # Access list in reverse order to have the newest items first.
+    for comment in reversed(c['comments']):
+        print("    {} - {}".format(comment['reviewer']['name'],
+                                   datetime.fromtimestamp(
+                                       comment["timestamp"])))
```
Comment (lengfeld) [inline repoload/repoload.py:241]: this code looks horrible. Maybe just introduce a variable to avoid this strange multipe line statement.

File: repoload/repoload.py [hunk 5]
```diff
@@ -151,13 +168,84 @@ def get_changes_for_topic(topic):
...
+    print('\n')
+    p = c["currentPatchSet"]
+    print("PatchSet\t{}:".format(p['number']))
+    print("Kind\t\t{}".format(p['kind']))
+    print("Ref\t\t{}".format(p['ref']))
+    print("Revision\t{}".format(p["revision"]))
+
+    parents = " \n".join(p["parents"])
+    print("Parents\t\t{}".format(parents))
+    print("Created\t\t{}".format(datetime.fromtimestamp(p["createdOn"])))
```
Comment (lengfeld) [inline repoload/repoload.py:219]: The above code looks very repetitive. I'm hesitating to ask whether there is a more elegant way to write it? Maybe it's worth a bit of indirection/abstraction.

File: repoload/repoload.py [hunk 6]
```diff
@@ -151,13 +168,84 @@ def get_changes_for_topic(topic):
...
+    if 'approvals' in p:
+        print("Approved by:")
+        for approval in p['approvals']:
+            print("    {} : {} [{}]".format(approval['value'],
+                                            approval['by']['name'],
+                                            approval['description']))
+
+    # TODO: Also show the comments on files
+    print('\n')
+    print('Comments:')
```
Comment (lengfeld) [inline repoload/repoload.py:232]: hmm, I think there should be maybe two separate commands or the command should accept a `--comment` argument. One show the basic information of a CR and one to show all information.
